internal/v2/certverifier/certverifier_test.go

```diff
@@ -155,40 +163,136 @@ func TestVerifyClientCertChain(t *testing.T) {
+		hostname                  string
+		rawCerts                  [][]byte
+		expectedErr               error
+		serverAuthorizationPolicy []byte
 	}{
 		{
-			description: "empty chain",
-			hostname:    "host",
-			rawCerts:    nil,
-			expectedErr: errors.New("server cert verification failed: server peer verification failed: server cert chain is empty"),
+			description:               "empty chain",
+			hostname:                  "host",
+			rawCerts:                  nil,
+			expectedErr:               errors.New("server cert verification failed: server peer verification failed: server cert chain is empty"),
+			serverAuthorizationPolicy: []byte(fakeServerAuthorizationPolicy),
 		},
 		{
-			description: "chain of length 1",
-			hostname:    "host",
-			rawCerts:    [][]byte{serverRootDERCert},
-			expectedErr: nil,
+			description:               "invalid server authorization policy",
+			hostname:                  "host",
+			rawCerts:                  [][]byte{serverRootDERCert},
+			expectedErr:               errors.New(fmt.Sprintf("rpc error: code = Unknown desc = server peer verification failed: invalid server authorization policy, expected: %s, got: .", fakeServerAuthorizationPolicy)),
+			serverAuthorizationPolicy: nil,
 		},
 		{
-			description: "chain of length 2 correct",
-			hostname:    "host",
-			rawCerts:    [][]byte{serverLeafDERCert, serverIntermediateDERCert},
-			expectedErr: nil,
+			description:               "chain of length 1",
+			hostname:                  "host",
+			rawCerts:                  [][]byte{serverRootDERCert},
+			expectedErr:               nil,
+			serverAuthorizationPolicy: []byte(fakeServerAuthorizationPolicy),
+		},
+		{
+			description:               "chain of length 2 correct",
+			hostname:                  "host",
+			rawCerts:                  [][]byte{serverLeafDERCert, serverIntermediateDERCert},
+			expectedErr:               nil,
+			serverAuthorizationPolicy: []byte(fakeServerAuthorizationPolicy),
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			// Create new stream to S2Av2.
+			opts := []grpc.DialOption{
+				grpc.WithTransportCredentials(insecure.NewCredentials()),
+				grpc.WithReturnConnectionError(),
+				grpc.WithBlock(),
+			}
+			conn, err := grpc.Dial(address, opts...)
+			if err != nil {
+				t.Fatalf("Client: failed to connect: %v", err)
+			}
+			defer conn.Close()
+			c := s2av2pb.NewS2AServiceClient(conn)
+			log.Printf("Client: connected to: %s", address)
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+			defer cancel()
+
+			// Setup bidrectional streaming session.
+			callOpts := []grpc.CallOption{}
+			cstream, err := c.SetUpSession(ctx, callOpts...)
+			if err != nil {
+				t.Fatalf("Client: failed to setup bidirectional streaming RPC session: %v", err)
+			}
+			log.Printf("Client: set up bidirectional streaming RPC session.")
+
+			// TODO(rmehta19): Add verificationMode to struct, and vary between tests.
```

Comment:
github shows this part as new code, but the TODO looks like from old code?

Reply:
Good catch! Removed.